lib/config-util.js

```diff
@@ -18,18 +18,45 @@ function guessCWD (cwd) {
   return cwd
 }
 
+function moduleExists (name) {
+  try {
+    return typeof require.resolve(name) !== 'undefined'
+  } catch (e) { return false }
+}
+
 Config.loadConfig = function (argv, cwd) {
   const rcOptions = [
     argv.nycrcPath || '.nycrc',
     '.nycrc.json',
+    '.nycrc.yml',
+    '.nycrc.yaml',
     'nyc.config.js'
   ]
   const rcPath = findUp.sync(rcOptions, { cwd: guessCWD(cwd) })
   let config = {}
 
   if (rcPath) {
-    if (rcPath.toLowerCase().endsWith('.js')) {
+    const rcPathLower = rcPath.toLowerCase()
+    if (rcPathLower.endsWith('.js')) {
       config = require(rcPath)
+    } else if (/.+\.y(a?)ml$/.test(rcPathLower)) {
+      const installedYamlParser = ['js-yaml', 'yaml'].find((name) => moduleExists(name))
+      switch (installedYamlParser) {
+        case 'js-yaml':
+          config = require('js-yaml').load(
+            fs.readFileSync(rcPath, 'utf8')
+          )
+          break
+        case 'yaml':
```

Comment:
Need to tell istanbul to ignore the `yaml` case as well since testing will always find js-yaml.